### PR TITLE
feat(HNG-53): created the waitlist hero card component

### LINKED ISSUE #53: [FEAT] Waitlist > Create the waitlist hero card
### Description
Develop a reusable Waitlist Card component. This component will be used to display the services on the waitlist hero section of the website. The Card should be visually appealing, responsive, and should include the following elements:

### Acceptance Criteria
The Waitlist Card component is implemented according to the design
The component should be fully responsive and look good on all screen sizes.
Ensure accessibility standards are met and proper semantic HTML.

### Requirements
The component should receive the following props

- Icon
- Title
- Description
- Testing

Ensure the component renders correctly with all required elements.
Check that the component handles different types of content (long titles, missing icons).

### Tasks
- [ ] Develop the waitlist hero cards component based on the provided requirements.
- [ ] Implement the component in the codebase.
- [ ] Write unit tests to ensure the component works as expected.
- [ ] Test the component for responsiveness and accessibility.
- [ ] Update documentation with usage instructions.

### Figma link
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=518-7653&m=dev

### Screenshot
waitlist-hero-card

### PR DESCRIPTION
**Pull request by judeb21**
# Description
Github Issue 53: Closes #53  

**Closes #53 **

# Changes proposed

## What were you told to do?
This PR is to create the wishlist hero card component

## What did you do?

- I created the waitlist hero card component
- I created a wishlist folder that will contain svg or images needed to complete the waitlist full page in the public directory
- I created a waitlist folder inside the common folder located in the component folder in the app directory
- I used [Shadcn card components](https://ui.shadcn.com/docs/components/card) in creating the card component and declared extra data types to accommodate the full card functionalities
- Created a waitlist card component test file inside a waitlist folder in the test folder of the app directory, where I wrote test cases to assert the functionality of the card component
- Added documentation in the contribution section of the readme file

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos

Screenshot 2024-07-19 at 01 51 11

### CODE REVIEW

**Review by JulianaSau — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -32,4 +32,37 @@ pnpm dev
...
+- OR
+
+```
+import WaitlistCard from '~/components/common/Waitlist/WaitlistCard';
+
+<WaitlistCard
+  cardTitle={"Easy Customization"}
+  cardDescription={cardDescription}
+  cardIcon={require("../../public/Wailtist/vector-icon.svg")}
+/>
```
Comment (JulianaSau) [inline README.md:63]: I dont think this documentation belongs here

File: package.json [hunk 2]
```diff
@@ -59,4 +59,4 @@
     "typescript": "^5",
     "vitest": "^1.6.0"
   }
-}
\ No newline at end of file
+}
```
Comment (JulianaSau) [inline package.json]: This change is not necessary. it's just a blank line. remove it

File: README.md (hunk 1 above)
Comment (judeb21) [reply README.md:63]: It's a contribution. I think it's good there or where do you think? @JulianaSau

File: package.json (hunk 2 above)
Comment (judeb21) [reply package.json]: Done @JulianaSau

**Comment by oreoluwa212** [comment]
Update your branch to match dev and make a PR asap please. ensure lint and test passes 
No unnecessary commenting

**Pull request closed without merging**